### PR TITLE
Added template code for ADGA dataset

### LINKED ISSUE #298: Feature Request:  Enhancing ADMIRALOPHTHA to include another dataset ADGA for "geographic atrophy"
### Feature Idea

1. Add ADGA for geographic atrophy with applicable parameters from the endpoints related to geographic atrophy indication
2. Update/Add the RMD/Vignette for ADG
3. Add Template code
4. Update Pharmaverse SDTM OE_OPHTA

### Relevant Input

Input will be similar to  ADBCVA with different  lookup table for parameters and aval/avalcat derivations

### Relevant Output

Output should have relevant parameters populated for the GA endpoints.

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by pavan-bhagat**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/cran-release/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update).
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files.
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions]
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately.
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiralophtha (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiralophtha site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralophtha/reference/index.html)" page.
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`.
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`.
- [ ] Link the issue so that it closes after successful merging.
- [ ] Address all merge conflicts and resolve appropriately.
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

To create the website for the content of this PR add "[create website]" to the title of the pull request.

### CODE REVIEW

File: inst/templates/ad_adga.R [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01A, TRT01P, STUDYEYE)
+
+adga_adslvar <- oe %>%
+  # Keep only GA related OE parameters
+    filter(
+    OETESTCD %in% c("AREA", "GAFLOC"),
+    !is.na(OESTRESC)
+    ) %>%
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:55]: Cleaner to do only in one line. But anyway, it seems strange to me to drop records with missing values, is that usual? Should we not deal with them in some way?
```suggestion
    filter(OETESTCD %in% c("AREA", "GAFLOC") & !is.na(OESTRESC)) %>%
```

File: inst/templates/ad_adga.R [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+# as needed and assign to the variables below.
+# For illustration purposes read in admiral test data
+
+data("oe_ophtha")
+data("admiral_adsl")
+
+# Add STUDYEYE to ADSL to simulate an ophtha dataset
+adsl <- admiral_adsl %>%
+  as.data.frame() %>%
+  mutate(STUDYEYE = sample(c("LEFT", "RIGHT"), n(), replace = TRUE)) %>%
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:26]: This is nice; please set a seed so that the same `adsl` is reproducible across different runs.

File: inst/templates/ad_adga.R [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      PARAMCD == "FSQRTFAF"  ~ "Fellow Eye Square Root Transformed GA Area measured by FAF(mm)"
+    ),
+    PARAMN = case_when(
+      PARAMCD == "SSQRTFAF"  ~ 5,
+      PARAMCD == "FSQRTFAF"  ~ 6
+    ),
+    AVAL = sqrt(aval),
+    AVALU = "mm"
+  ) %>%
+  select(-aval)
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:123]: Please implement these derivations using `derive_param_computed()`. if you have two similar calls you could consider using `call_derivation()` as well. You can read [here](https://pharmaverse.github.io/admiral/cran-release/reference/derive_param_computed.html?q=derive_param_computed#null) and [here](https://pharmaverse.github.io/admiral/cran-release/articles/higher_order.html#call-derivation) for details. I recommend giving it a go yourself first, but you could also try using the Ask AI widget on the admiral website to convert your derivaton to a more "admiral"-style call.

File: inst/templates/ad_adga.R [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  filter(PARAMCD %in% c("SAREAFAF", "FAREAFAF")) %>%
+  rename(aval = AVAL) %>%
+  mutate(
+    PARAMCD = case_when(
+      PARAMCD == "SAREAFAF"  ~ "SSQRTFAF",
+      PARAMCD == "FAREAFAF"  ~ "FSQRTFAF"
+    ),
+    PARAM = case_when(
+      PARAMCD == "SSQRTFAF"  ~ "Study Eye Square Root Transformed GA Area measured by FAF(mm)",
+      PARAMCD == "FSQRTFAF"  ~ "Fellow Eye Square Root Transformed GA Area measured by FAF(mm)"
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:114]: Missing spaces before the units?

File: inst/templates/ad_adga.R [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    AVALU = "mm"
+  ) %>%
+  select(-aval)
+
+# Combine all parameters
+adga_comb <- bind_rows(
+  adga_visit,
+  adga_derive
+) %>%
+  arrange(USUBJID, PARAMN, AVISITN)
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:130]: You won't need this anymore if you use `derive_param_computed()` as above.

File: inst/templates/ad_adga.R [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+  # If worst result is lowest result, change mode to "first"
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD, ATPT)),
+      order = exprs(AVAL, ADT),
+      new_var = WORS01FL,
+      mode = "last"
+    ),
+    filter = !is.na(AVISITN) & (ONTRTFL == "Y" | ABLFL == "Y") & PARAMCD %in% c("SSQRTFAF", "FSQRTFAF")
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:187]: Is there no way to assess what's "worst" for these parameters? If so it's ok, but please leave a comment explaining why.

File: inst/templates/ad_adga.R [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+  )
+
+# Derive baseline information
+adga_change <- adga_vstflag %>%
+  # Calculate BASE (do not derive for GA Lesion Focality params)
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:192]: how come?

File: inst/templates/ad_adga.R [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+
+# Save output ----
+
+dir <- tools::R_user_dir("admiralophtha_templates_data", which = "cache")
+# Change to whichever directory you want to save the dataset in
+if (!file.exists(dir)) {
+  # Create the folder
+  dir.create(dir, recursive = TRUE, showWarnings = FALSE)
+}
+save(admiralophtha_adga, file = file.path(dir, "adga.rda"), compress = "bzip2")
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:247]: You can temporarily edit this line to: 
```
save(admiralophtha_adga, file = file.path("data", "admiralophtha_adga.rda"), compress = "bzip2")
```
to save the dataset.

**Review by manciniedoardo — COMMENTED** [review]
Thanks @pavan-bhagat it's a great start! Left you some initial comments. For your next iteration please also make sure to go through the PR checklist as that will contain all the items which you need to cover, e.g. updating the news file or documenting the data.

**Comment by manciniedoardo** [comment]
Hi @pavan-bhagat how is this going?

File: inst/templates/ad_adga.R (hunk 1 above)
Comment (pavan-bhagat) [reply inst/templates/ad_adga.R]: this has been updated  with the latest update from PHARMAVERSESDTM, there are no missing results for AREA anymore

File: inst/templates/ad_adga.R (hunk 2 above)
Comment (pavan-bhagat) [reply inst/templates/ad_adga.R:29]: Added set.seed(1234)

File: inst/templates/ad_adga.R (hunk 8 above)
Comment (pavan-bhagat) [reply inst/templates/ad_adga.R:262]: thank you, applied the above suggetion!

File: inst/templates/ad_adga.R (hunk 6 above)
Comment (pavan-bhagat) [reply inst/templates/ad_adga.R]: Highest result is the worst result, hence the mode is assigned as Last

File: inst/templates/ad_adga.R (hunk 7 above)
Comment (pavan-bhagat) [reply inst/templates/ad_adga.R:207]: BASE is not derrived for GA lesion focality as the results are charecter (S, NS), hence BASE will be missing, however BASEC is populated

File: inst/templates/ad_adga.R (hunk 3 above)
Comment (pavan-bhagat) [reply inst/templates/ad_adga.R]: implemented derivation using derive_param_computed()

File: inst/templates/ad_adga.R (hunk 5 above)
Comment (pavan-bhagat) [reply inst/templates/ad_adga.R]: removed this snipped after derive_param_computed()

File: inst/templates/ad_adga.R [hunk 9]
```diff
@@ -0,0 +1,264 @@
+# Name: ADGA
+#
+# Label: Geographic Atrophy  Analysis Dataset
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R]:
```suggestion
# Label: Geographic Atrophy Analysis Dataset
```

File: inst/templates/ad_adga.R [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+  # Derive AFEYE needed for PARAMCD derivation
+  derive_var_afeye(loc_var = OELOC, lat_var = OELAT, loc_vals = c("EYE", "RETINA"))
+
+adga_param <- adga_aval %>%
+  # Add PARAM, PARAMCD, PARAMN
+  derive_vars_merged(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAM, PARAMCD, PARAMN),
+    by_vars = exprs(OETESTCD, AFEYE)
+  ) %>%
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R]: It's not wrong to use `derive_vars_merged()` here, but `{admiral}` provides a more specialised function `derive_vars_merged_lookup()` which we should use instead as it can tell you if some records are unmapped. I noticed the other templates also don't use it and made an issue to change that - see [here](https://github.com/pharmaverse/admiralophtha/issues/311).


```suggestion
  derive_vars_merged_lookup(
    dataset_add = param_lookup,
    new_vars = exprs(PARAM, PARAMCD, PARAMN),
    by_vars = exprs(OETESTCD, AFEYE)
  ) %>%
```

File: inst/templates/ad_adga.R [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+      get_admiral_option("subject_keys"),
+      exprs(VISIT, VISITNUM, OEDY, OEDTC, AFEYE, !!!adsl_vars)
+    ),
+    variable_params = list(
+      # Study eye
+      params(
+        parameters = exprs(
+          # Users may need to update this code to identify the correct records to use.
+          SESQRT = PARAMCD == "SAREAFAF"
+        ),
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R]: If only one thing inside `exprs()`, we prefer to format over one line.
```suggestion
        # Users may need to update this code to identify the correct records to use.
        parameters = exprs(SESQRT = PARAMCD == "SAREAFAF"),
```

File: inst/templates/ad_adga.R [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+          AVALC = as.character(AVAL),
+          AVALU = "mm",
+        )
+      ),
+      # Fellow eye
+      params(
+        parameters = exprs(
+          # Users may need to update this code to identify the correct records to use.
+          FESQRT = PARAMCD == "FAREAFAF"
+        ),
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R]: If only one thing inside `exprs()`, we prefer to format over one line.
```suggestion
         # Users may need to update this code to identify the correct records to use.
        parameters = exprs(FESQRT = PARAMCD == "FAREAFAF"),
```

File: inst/templates/ad_adga.R (hunk 7 above)
Comment (manciniedoardo) [reply inst/templates/ad_adga.R:207]: ok, please ensure to encapsulate this info when you come to write the `ADGA` vignette.

File: inst/templates/ad_adga.R [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+      exprs(VISIT, VISITNUM, OEDY, OEDTC, AFEYE, !!!adsl_vars)
+    ),
+    variable_params = list(
+      # Study eye
+      params(
+        parameters = exprs(
+          # Users may need to update this code to identify the correct records to use.
+          SESQRT = PARAMCD == "SAREAFAF"
+        ),
+        set_values_to = exprs(
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R:99]: Could/should you set `DYPE = "SQRT"` here and below?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Just a couple more comments @pavan-bhagat but it's looking good!

File: inst/templates/ad_adga.R [hunk 14]
```diff
@@ -0,0 +1,264 @@
...
+
+# Save output ----
+
+dir <- tools::R_user_dir("admiralophtha_templates_data", which = "cache")
+# Change to whichever directory you want to save the dataset in
+if (!file.exists(dir)) {
+  # Create the folder
+  dir.create(dir, recursive = TRUE, showWarnings = FALSE)
+}
+save(admiralophtha_adga, file = file.path("data", "admiralophtha_adga.rda"), compress = "bzip2")
```
Comment (manciniedoardo) [inline inst/templates/ad_adga.R]: Please revert this back to the original temporary directory, like in other templates, when you are done.

Comment (pavan-bhagat) [reply inst/templates/ad_adga.R]: Thank you, applied all the suggetions!

**Review by manciniedoardo — APPROVED** [review]
Great job @pavan-bhagat!
